controllers/machinedeployment_controller.go

```diff
@@ -199,6 +199,10 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, cluster *cl
 		return ctrl.Result{}, r.rolloutRolling(ctx, d, msList)
 	}
 
+	if d.Spec.Strategy.Type == clusterv1.OnDeleteMachineDeploymentStrategyType {
```

Comment:
Let's update https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha4/machinedeployment_types.go#L104-L108, specifically the part that says `Currently the only supported strategy is
	// "RollingUpdate".`

Reply:
Is there any kind of webhook validation that makes sure the `Strategy.Type` is valid?

Reply:
There is no webhook validation on ensuring type is valid

Reply:
The reconciliation loop will error through if it's an unknown strategy.

Reply:
It would probably be good to add an annotation to do Enum validation on the field, that way it's more clear to users that there is an issue: https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
One downside here is that if someone already has a resource created with an invalid strategy, introducing enum-based validation here would result in any attempts to reconcile that resource without updating the value fail where they weren't previously failing, but I'm not sure that's too big of an issue since the resource wouldn't be properly reconciling anyway.

Not sure if @vincepri or @CecileRobertMichon have thoughts on this

Reply:
Happy to add that if y'all like!

Reply:
I'm +1 adding the enum-based validation now as part of the v0.4 release

Reply:
+1 for adding an enum validation